examples/mnist/train_mnist.py

```diff
@@ -44,8 +44,6 @@ def main():
                         'negative integer, NumPy arrays are used')
     parser.add_argument('--out', '-o', default='result',
                         help='Directory to output the result')
-    parser.add_argument('--resume', '-r', type=str,
```

Comment:
Should we remove this argument?
Some users might want to control the resume behavior.
I think we should give users a choice from the following:
* Resume from a specific snapshot (`--resume`).
* Enable `autoload` and load the latest snapshot if available.
* Disable `autoload` and always start a new training.

So, how about retaining `--resume` argument, and introducing another argument to enable `autoload` feature?